src/fmu/dataio/providers/_objectdata.py

```diff
@@ -323,10 +323,10 @@ def get_spec(self) -> dict[str, Any]:
             exclude_none=True,
         )
 
-    def get_bbox(self) -> dict[str, Any]:
+    def get_bbox(self) -> None:
```

Comment:
I think it's preferable to keep the interface for these abstract methods consistent such that every implementation of ` ObjectDataProvider` has the same method signature (returns the same type). An easy way around that is just doing

```python
spec=self.get_spec() or None,
bbox=self.get_bbox() or None,
```

where the `DerivedObjectDescriptor` is returned -- and empty dict returned from those methods will evaluate falsey.